### PR TITLE
Allow disable mcac even when CDC enabled

### LINKED ISSUE #481: Enabling CDC messes up mgmt-api configuration
**What happened?**
management-api uses properties to handle how the JVM_OPTS is modified. Enabling CDC messes up this behavior and causes things that are not supposed to be there to re-appear or go missing. For example, MGMT_API_DISABLE_MCAC should not cause MCAC to appear.

The entire generate_config.go should probably be history at some point and the configuration of the startup properties should happen in the mgmt-api repository. 

**Did you expect to see something different?**

**How to reproduce it (as minimally and precisely as possible)**:

**Environment**

* Cass Operator version:

    `Insert image tag or Git SHA here`
    
    

* Kubernetes version information:

    `kubectl version`
    

* Kubernetes cluster kind:

    insert how you created your cluster: kops, bootkube, etc.

* Manifests:

```
insert manifests relevant to the issue
```

* Cass Operator Logs:

```
insert Cass Operator logs relevant to the issue here
```

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by Miles-Garnsey**
**What this PR does**:

Ensures that CDC classpath/agent manipulation logic doesn't erroneously re-enable MCAC when the MCAC environment variable is set to disable it.

**Which issue(s) this PR fixes**:
Fixes #481 

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [ ] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by Miles-Garnsey** [comment]
This appears to be good to go now, there was one mistake and some test flakes happening earlier, but I think it is all good now.

**Review by emerkle826 — APPROVED** [review]

File: pkg/cdc/generate_config.go [hunk 1]
```diff
@@ -90,17 +91,46 @@ func updateAdditionalJVMOpts(optsSlice []string, CDCConfig *cassdcapi.CDCConfigu
 			optsSlice = append(optsSlice, nameTag+"="+fmt.Sprintf("%s", reflectedValue))
 		}
 	}
+	// We need to add these two elements to the slice first, because the management agent must start before the CDC agent.
```
Comment (olim7t) [inline pkg/cdc/generate_config.go]: Is this comment misplaced? It looks like a duplicate of the one on line 103.

File: pkg/cdc/generate_config.go [hunk 2]
```diff
@@ -90,17 +91,46 @@ func updateAdditionalJVMOpts(optsSlice []string, CDCConfig *cassdcapi.CDCConfigu
...
+	// If MCAC is disabled, we need to avoid adding it to startup.
```
Comment (olim7t) [inline pkg/cdc/generate_config.go]: Same here, this refers to MCAC but the next line deals with CDC. Shouldn't this be merged with line 97 rather?

File: pkg/cdc/generate_config.go (hunk 1 above)
Comment (Miles-Garnsey) [reply pkg/cdc/generate_config.go]: Probably, this is why some people say don't write comments - they fall out of date :/ 

I'm not sure what my thinking was there so I'm getting rid of these.

File: pkg/cdc/generate_config.go (hunk 2 above)
Comment (Miles-Garnsey) [reply pkg/cdc/generate_config.go]: Ditto here.

**Comment by burmanm** [comment]
I'll fix the missing CHANGELOG line before release.

**Pull request merged**